process/block/preprocess/transactions.go

```diff
@@ -961,7 +961,12 @@ func (txs *transactions) createAndProcessMiniBlocksFromMe(
 		}
 
 		miniBlock.TxHashes = append(miniBlock.TxHashes, txHash)
-		txs.blockSizeComputation.AddNumTxs(1)
+		numTxs := 1
+		if core.IsSmartContractAddress(tx.RcvAddr) {
```

Comment:
this is not the correct way to resolve it. Add blockSizeComputation component to smart contract result post processor and call AddNumTxs there when new cross shard SCRs are generated. It is not ensured that one tx = one scr

Reply:
When we reach the postprocessor it is already too late as the transactions are already executed and we need to include the scr(s) as well. Indeed, it is not ensured that one tx = one scr but at least it will estimate better.

Reply:
Actually you will have nothing in post processor when you create a miniblock from me to meta with 27k stake systems SC calls. In fact at the destination, in this case in meta, could result in 54k txs (from which 27k are SCR). Because you need to process at least one mb completely this could result in a stuck situation. So, you also need to multiply at source, with 2 or more (in terms of size, depending of how many SCR could exist), all the SC calls when you create mbs from me. I think that this situation on normal cases should be solved by gas limit and in case of an invalid SC call, adding only one additional tx which simulates the scr created at destination, should be enough as in this case would be only one SCR for each invalid SC call?

Reply:
Added block size computation taking in consideration also the post process mbs and txs when creating mbs dest me